### PR TITLE
freetype: Add option with_brotli

### PR DESCRIPTION
**Pull request by kyllingstad**
Specify library name and version:  **freetype/2.10.x**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This adds support for the `FT_WITH_BROTLI` configuration macro in FreeType, which was [added in version 2.10.2](https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/CMakeLists.txt?h=VER-2-10-2#n71).

By default, FreeType's build system tries to be clever about which features are enabled: If it can detect an optional dependency on the system, it will enable it, otherwise it will be disabled. Crucially, *the search is not limited to the Conan package tree*; it also searches in `/usr` and could pick up `libbrotli` there. (That happened in my case, leading to linker errors down the line because the Brotli version on my system was incompatible with the FreeType version from conan-center.) Therefore, it is important that all of the optional dependencies are included in the conanfile.

### CODE REVIEW

File: recipes/freetype/all/conanfile.py [hunk 1]
```diff
@@ -47,6 +49,8 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if tools.Version(self.version) < "2.10.2":
+            del self.options.with_brotli
```
Comment (SpaceIm) [inline recipes/freetype/all/conanfile.py:53]: I think that it's better to remove options based on version in `config_option`.

File: recipes/freetype/all/conanfile.py [hunk 2]
```diff
@@ -57,6 +61,8 @@ def requirements(self):
             self.requires("zlib/1.2.11")
         if self.options.with_bzip2:
             self.requires("bzip2/1.0.8")
+        if tools.Version(self.version) >= "2.10.2" and self.options.with_brotli:
```
Comment (SpaceIm) [inline recipes/freetype/all/conanfile.py:64]:
```suggestion
        if self.options.get_safe("with_brotli"):
```

**Comment by kyllingstad** [comment]
Thanks for the review! I've implemented the changes you suggested, and also changed the test to `if "with_brotli" in self.options` in the `_configure_cmake()` function.

**Comment by SpaceIm** [comment]
> By default, FreeType's build system tries to be clever about which features are enabled: If it can detect an optional dependency on the system, it will enable it, otherwise it will be disabled. Crucially, the search is not limited to the Conan package tree; it also searches in /usr and could pick up libbrotli there. (That happened in my case, leading to linker errors down the line because the Brotli version on my system was incompatible with the FreeType version from conan-center.) Therefore, it is important that all of the optional dependencies are included in the conanfile.

Is it possible to add a patch for Freetype in order to honor recipe options when they are disabled? I understand that it can still fail to properly not using a dependency if disabled in options.

**Comment by kyllingstad** [comment]
The problem is not that recipe options aren't honored. You can explicitly enable a specific dependency/feature X using the CMake variable `FT_WITH_X`, or you can explicitly disable it using `CMAKE_DISABLE_FIND_PACKAGE_X`. That's how the current recipe options are implemented.

But if you do neither of the above, the dependency/feature will be enabled or disabled depending on whether it exists on the host system. Since those CMake variables are associated with individual, named dependencies, it is not possible within the current system to guard against future additions. There is no catch-all variable to disable autodetection of dependencies altogether.

Furthermore, I cannot see a way to add one without making quite significant changes to the upstream `CMakeLists.txt`. When the FreeType developers add a new optional dependency, I think that the Conan recipe simply has to follow suit. That seems like the least fragile solution to me.

**Comment by ledocc** [comment]
@kyllingstad please request Early Access Program to run build, and allow this PR to be reviewed and approved.
Good job BTW.

**Review by ericLemanissier — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@kyllingstad please request EAP on https://github.com/conan-io/conan-center-index/issues/4

**Comment by kyllingstad** [comment]
I've requested EAP access now.

File: recipes/freetype/all/conanfile.py [hunk 3]
```diff
@@ -57,6 +61,8 @@ def requirements(self):
             self.requires("zlib/1.2.11")
         if self.options.with_bzip2:
             self.requires("bzip2/1.0.8")
+        if self.options.get_safe("with_brotli"):
+            self.requires("brotli/1.0.7")
```
Comment (SpaceIm) [inline recipes/freetype/all/conanfile.py:65]:
```suggestion
            self.requires("brotli/1.0.9")
```

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**